PyInstaller/building/api.py

```diff
@@ -381,7 +382,14 @@ def __init__(self, *args, **kwargs):
             if isinstance(arg, TOC):
                 self.toc.extend(arg)
             elif isinstance(arg, Target):
-                self.toc.append((os.path.basename(arg.name), arg.name, arg.typ))
+                if hasattr(arg, '_EXE'):
```

Comment:
Also, this logic doesn't work for target dependencies, but that should be fixed in the future.